pkgs/tools/security/gnupg/21.nix

```diff
@@ -29,7 +29,8 @@ stdenv.mkDerivation rec {
     sed -i 's,"libpcsclite\.so[^"]*","${pcsclite}/lib/libpcsclite.so",g' scd/scdaemon.c
   ''; #" fix Emacs syntax highlighting :-(
 
-  configureFlags = optional x11Support "--with-pinentry-pgm=${pinentry}/bin/pinentry";
+  gpgPinentryBinaryPath = pinentry.gpgPinentryBinaryPath or "bin/pinentry";
+  configureFlags = optional (pinentry != null) "--with-pinentry-pgm=${pinentry}/${gpgPinentryBinaryPath}";
```

Comment:
I think it should be pretty obvious that the original objective of
`x11Support` option is to drop dependencies on x11 libs on headless
systems. E.g. gpg can be used in batch mode to verify signatures, and
that doesn't require either of x11 libs or pinentry.